src/api/professions.js

```diff
@@ -0,0 +1,66 @@
+const express = require("express");
+const professionsExample = require("./dev/professionsExample");
+
+const router = express.Router();
+
+router.get("/", (req, res) => {
+  res.status(200).json(professionsExample);
+});
+
+router.get("/:id", (req, res) => {
+  const id = +req.params.id;
+  const result = professionsExample.find((profession) => profession.id === id);
+  if (result) res.status(200).json(result);
+  else
+    res.status(404).json({
+      message: "No profession found",
+    });
+});
+
+router.post("/", (req, res) => {
+  const initial = professionsExample;
+  const newProfession = {
+    id: [...initial].pop().id + 1,
+    ...req.body,
+  };
+  initial.push(newProfession);
+  res.status(201).json(initial[initial.length - 1]);
+});
+
+router.put("/:id", (req, res) => {
+  const id = +req.params.id;
+  const index = professionsExample.findIndex(
+    (profession) => profession.id === id
+  );
+  if (index >= 0) {
+    let elementToUpdate = professionsExample[index];
+    elementToUpdate = {
+      id,
+      ...req.body,
```

Comment:
If the body contain only modified fields, original datas will be deleted (have to do like companies path)